astropy/convolution/src/convolve.c

```diff
@@ -159,6 +199,7 @@ FORCE_INLINE void convolve1d_boundary_none(DTYPE * const result,
 #endif
 
     const size_t _wkx = _nkx / 2;
+
 #ifdef NDEBUG
```

Comment:
I think the conditions below need to be conditional on ``_padded``, no? In the padded case, it is implementable to have a kernel larger than the image. Might not make sense to do so, but that's the user's prerogative.